client/ayon_core/plugins/publish/extract_otio_review.py

```diff
@@ -105,42 +107,48 @@ def process(self, instance):
             otio_media = r_otio_cl.media_reference
             media_metadata = otio_media.metadata
 
+            variables = (self.to_width, self.to_height)
+            keys = ("width", "height")
+
             # get from media reference metadata source
             # TODO 'openpype' prefix should be removed (added 24/09/03)
             # NOTE it looks like it is set only in hiero integration
-            for key in {"ayon.source.width", "openpype.source.width"}:
-                value = media_metadata.get(key)
-                if value is not None:
-                    width = int(value)
-                    break
-
-            for key in {"ayon.source.height", "openpype.source.height"}:
-                value = media_metadata.get(key)
-                if value is not None:
-                    height = int(value)
-                    break
-
-            # compare and reset
-            if width != self.to_width:
-                self.to_width = width
-            if height != self.to_height:
-                self.to_height = height
+            for variable, key in zip(variables, keys):
```

Comment:
Maybe more clear?
```suggestion
            for variable, key in (
                (self.to_width: "width"),
                (self.to_height: "height"),
            ):
```

Reply:
Did you mean:
```suggestion
            for variable, key in (
                (self.to_width, "width"),
                (self.to_height, "height"),
            ):
```

Or is your code actually valid code? 😕

Reply:
Also, if `height != self.to_height:` and then setting `self.to_height = height`. Isn't that just the same as always doing:
```
self.to_height = height
```

Reply:
Sorry @BigRoy not sure I'm following you.. 

The logic here is get `self.to_height`/`self.to_width` initialized with `instance.data` or default (1280x720). Then optionally overwrite it with data found in media_metadata. 

was that your question ?

Reply:
Github just showed me this old code on the side:
![image](https://github.com/user-attachments/assets/90d3bfe0-c67c-42e9-9299-b4d13b663637)

And I figured, isn't that just _always_ setting it? But maybe it's not relevant to the new code.

Reply:
Oh got it, well yeah new code edits it as soon as `media_metadata.get(meta_key)` exists now.

Reply:
I didn't look at the full code, but it does not make sense either. You're storing the value to variable `variable` so you won't have the value available. Wouldn't that be this?
```python
                output = {"width": self.to_width, "height": self.to_height}
                for meta_prefix in ("ayon.source.", "openpype.source."):
                    for key in ("width", "height"):
                        meta_key = f"{meta_prefix}.{key}"
                        if media_metadata.get(meta_key):
                            output[key] = media_metadata[meta_key]
                            break

                width = output["width"]
                height = output["height"]
```

Reply:
You are totally right, I don't know what I was thinking. This is adjusted.